Makefile

```diff
@@ -564,6 +565,7 @@ $(eval $(call add_define,USE_COHERENT_MEM))
 $(eval $(call add_define,USE_TBBR_DEFS))
 $(eval $(call add_define,WARMBOOT_ENABLE_DCACHE_EARLY))
 $(eval $(call add_define,BL2_AT_EL3))
+$(eval $(call add_define,BL2_IN_XIP_MEM))
```

Comment:
I don't see the check I suggested in my last review: 
"... you could avoid supporting the case where BL2_AT_EL3 = 0 and BL2_IN_XIP_MEM = 1, by throwing an error in the top level Makefile (in the "Check incompatible options" section)"

Reply:
Sorry, I thought that we need to add build check in platform makefile, will add it to top Makefile, thanks.

Reply:
Hi, Dan, have updated the patch according to your comments, please review again, thanks.